pypesto/optimize/ess/sacess.py

```diff
@@ -700,90 +701,166 @@ def get_default_ess_options(num_workers: int, dim: int) -> List[Dict]:
     Setting appropriate values for ``n_threads`` and ``local_optimizer`` is
     left to the user. Defaults to single-threaded and no local optimizer.
 
+    Based on https://bitbucket.org/DavidPenas/sacess-library/src/508e7ac15579104731cf1f8c3969960c6e72b872/src/method_module_fortran/eSS/parallelscattersearchfunctions.f90#lines-929
+
     Parameters
     ----------
     num_workers: Number of configurations to return.
     dim: Problem dimension.
     """
-    min_dimrefset = 3
+    min_dimrefset = 5
+
+    def dim_refset(x):
+        return max(min_dimrefset, ceil((1 + sqrt(4 * dim * x)) / 2))
```

Comment:
Is this also based on [the reference](https://bitbucket.org/DavidPenas/sacess-library/src/508e7ac15579104731cf1f8c3969960c6e72b872/src/method_module_fortran/eSS/parallelscattersearchfunctions.f90#lines-929). Could not find the function `MOD(ids...)`

Reply:
Yes. The `MOD` is replaced by `cycle` at the end of this function.